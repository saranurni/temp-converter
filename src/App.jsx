import React from "react";
import "./App.css";
import Input from "./components/Input/Input";
import tryConvert from "./lib/tryConvert";

function App() {
  const [values, setValues] = React.useState({ celsius: 0, fahrenheit: 0 });

  const handleChange = (event) => {
    if (event.target.id === "celsius") {
      setValues({
        ...values,
        celsius: event.target.value,
        fahrenheit: tryConvert(event.target.value, "toFahrenheit"),
      });
    } else {
      setValues({
        ...values,
        fahrenheit: event.target.value,
        celsius: tryConvert(event.target.value, "toCelsius"),
      });
    }
  };
  return (
    <>
      <h1 className="my-4 text-center text-2xl font-bold">
        Temperature Converter
      </h1>

      <main>
        <form className="mx-auto mt-8 flex w-1/3 justify-between">
          <Input
            label="Celsius"
            value={Number(values.celsius)}
            changeHandler={handleChange}
          />
          <Input
            label="Fahrenheit"
            value={Number(values.fahrenheit)}
            changeHandler={handleChange}
          />
        </form>
      </main>
    </>
  );
}
export default App;
