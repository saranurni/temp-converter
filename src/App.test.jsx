import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import App from "./App";

it("renders without crashing", () => {
  const rendered = renderer.create(<App />).toJSON();

  expect(rendered).toMatchSnapshot();
});

describe("App", () => {
  it("converts Celsius to Fahrenheit", () => {
    render(<App />);

    const input = screen.getByLabelText(/celsius/i);
    userEvent.type(input, "100");

    expect(screen.getByLabelText(/fahrenheit/i).value).toBe("212");
  });
  it("converts Fahrenheit to Celsius", () => {
    render(<App />);

    const input = screen.getByLabelText(/fahrenheit/i);
    userEvent.type(input, "212");

    expect(screen.getByLabelText(/celsius/i).value).toBe("100");
  });
});
