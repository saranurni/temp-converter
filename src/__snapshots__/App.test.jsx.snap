// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <h1
    className="my-4 text-center text-2xl font-bold"
  >
    Temperature Converter
  </h1>,
  <main>
    <form
      className="mx-auto mt-8 flex w-1/3 justify-between"
    >
      <div
        className="flex flex-col gap-y-1"
      >
        <label
          htmlFor="celsius"
        >
          Celsius
        </label>
        <input
          className="w-36 rounded border"
          id="celsius"
          onChange={[Function]}
          type="number"
          value={0}
        />
      </div>
      <div
        className="flex flex-col gap-y-1"
      >
        <label
          htmlFor="fahrenheit"
        >
          Fahrenheit
        </label>
        <input
          className="w-36 rounded border"
          id="fahrenheit"
          onChange={[Function]}
          type="number"
          value={0}
        />
      </div>
    </form>
  </main>,
]
`;
